### DISCUSSION TITLE
an error occurs while use both official kernels and costum kernels

### DISCUSSION THREAD

**Discussion by bdliangxy** [idaholab/moose#25255] [Q&A General] [answered]
Hi! All!
I've recently learned MOOSE for just a few days. i'v noticed a paper , 
[Jin 等 - 2022 - Protocol for phase-field simulations of lithium de.pdf](https://github.com/idaholab/moose/files/12397092/Jin.-.2022.-.Protocol.for.phase-field.simulations.of.lithium.de.pdf)
made a instruction in detail on Li dendrite simulation by MOOSE. I followed the protocol to the letter, but a few issues blocked me.

Something wrong happened in the 7th step. I need to put the *.C and *.h files provided by aurthor in the phase_field/src/kernels and phase_field/include/kernels folders, respectively and then recomplie it. The code is here

[code.zip](https://github.com/idaholab/moose/files/12397107/code.zip)

I'm confused about which directory I should put in.I put them in the corresponding position in the official phase_field directory `~/projects/moose/modules/phase_field` or put them in the corresponding position in the directory made by stork.sh，an error always was shown like:
```
(moose) liangxy@liangxy:~/projects/moose/modules/phase_field$ make -j10
Creating Unity /home/liangxy/projects/moose/modules/phase_field/build/unity_src/kernels_Unity.C
Compiling C++ (in opt mode) /home/liangxy/projects/moose/modules/phase_field/build/unity_src/kernels_Unity.C...
Linking Library /home/liangxy/projects/moose/modules/phase_field/lib/libphase_field-opt.la...
Linking Executable /home/liangxy/projects/moose/modules/phase_field/phase_field-opt...
/home/liangxy/mambaforge3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/liangxy/projects/moose/modules/phase_field/lib/libphase_field-opt.so: undefined reference to `InputParameters validParams<Kernel>()'
collect2: error: ld returned 1 exit status
make: *** [/home/liangxy/projects/moose/framework/app.mk:445：/home/liangxy/projects/moose/modules/phase_field/phase_field-opt] 错误 1
```
I consulted the author , and he said it's problem concrning register, but no further response. So I add `registerMooseObject("phase_fieldApp", Migration);`, `registerMooseObject("phase_fieldApp", Kinetics);`,  `registerMooseObject("phase_fieldApp", Conduction);` in the corresponding .C files . Modified kernel files again put in the official modules directory or stork.sh made directory. The same error always appears.
Anyone please help me. Thanks in advance!

**Comment by GiudGiud** [comment, marked as answer]
See these instructions to remove this deprecated syntax:
https://mooseframework.inl.gov/newsletter/2021/2021_11.html#legacy-input-parameter-deprecation

**Reply by bdliangxy** [reply]
Thank you a lot. Perferct answer and replied almost instantly. It bothered me for a week and you are my God!
